PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/CompareSubgraphs.java

```diff
@@ -0,0 +1,257 @@
+     * @return the ordered subgraph.
+     */
+    private static ArrayList<GraphNodeOrder> orderNodes(
+        HashMap<Integer, GraphNodeOrder> mainGraphOrder, OriginalGraph subGraph) {
+      ArrayList<GraphNodeOrder> subGraphOrder = new ArrayList<>();
+      subGraph.getNodes().forEach((Integer id, Node node) -> {
+        int level = mainGraphOrder.get(id).getLevel();
+        subGraphOrder.add(new GraphNodeOrder(node, level));
+      });
+      return subGraphOrder;
+    }
+
+    /**
+     * Mark all of the overlapping nodes in the subgraph.
+     *
+     * @param overlappingNodes a map containing all of the overlapping nodes.
+     * @param nodeOrder        the ordered list of nodes of the subgraph.
+     */
+    private static void markOverlap(HashMap<Integer, AbstractNode> overlappingNodes,
+        ArrayList<GraphNodeOrder> nodeOrder) {
+      nodeOrder.forEach(graphNode -> {
+        graphNode.setOverlapping(overlappingNodes.containsKey(
+            graphNode.getNode().getId()));
+      });
+    }
+
+    /**
+     * Calculate the subgraph order and mark the overlapping nodes.
+     */
+    @Override
+    public void run() {
+      nodeOrder = orderNodes(mainGraphOrder, subGraph);
+      Collections.sort(nodeOrder);
+      try {
+        overlapThread.join();
+        markOverlap(overlapThread.overlappedNodes, nodeOrder);
+      } catch (InterruptedException ex) {
+        Logger.getLogger(CompareSubgraphs.class.getName()).log(Level.SEVERE, null, ex);
+      }
+    }
+  }
+
+  /**
+   * Thread which can be used to calculate the overlapping nodes of two graph.
+   */
+  private static class OverlapThread extends Thread {
+
+    private HashMap<Integer, AbstractNode> overlappedNodes;
+    private final OriginalGraph firstGraph;
+    private final OriginalGraph secondGraph;
+
+    /**
+     * Construct a overlap thread, which finds the overlapping nodes of the given graphs.
+     *
+     * @param firstGraph  the first graph.
+     * @param secondGraph the second graph.
+     */
+    private OverlapThread(OriginalGraph firstGraph, OriginalGraph secondGraph) {
+      this.firstGraph = firstGraph;
+      this.secondGraph = secondGraph;
+    }
+
+    /**
+     * Wait till the thread completes its execution and get the overlapping nodes of the graphs.
+     *
+     * @return the overlapping nodes of the graphs.
+     */
+    public HashMap<Integer, AbstractNode> getOverlappedNodes() {
+      try {
+        this.join();
+      } catch (InterruptedException ex) {
+        Logger.getLogger(CompareSubgraphs.class.getName()).log(Level.SEVERE, null, ex);
+      }
+      return overlappedNodes;
+    }
+
+    /**
+     * Get all of the nodes which are in both the top graph and the bottom graph.
+     *
+     * @param topGraph    the top graph.
+     * @param bottomGraph the bottom graph.
+     * @return the overlapping nodes.
+     */
+    private static HashMap<Integer, AbstractNode> calculateOverlappedNodes(OriginalGraph topGraph,
```

Comment:
First it was defined outside of this nested class, after refactoring I forgot to change it to non-static. Fixed.